### ISSUE TITLE
How to export go graphql schemas as graphql schema file?

### ISSUE THREAD

**Issue by FrontMage** [graphql-go/graphql#353]
Wonder if there is a way to export schemas or just have to write it twice.

**Comment by ccbrown** [comment]
I don't think this library has built-in support for exporting GraphQL schema files, but you definitely don't need to write your schema twice. There are tools that can generate schemas from running servers. So for example, if you install [graphql-cli](https://github.com/graphql-cli/graphql-cli), you can run `graphql init`, give it a project name and the endpoint for your server, then run `graphql get-schema` to get a GraphQL schema language file for your server.

**Comment by FrontMage** [comment]
@ccbrown Thanks for the advice. But the `graphql-cli` is using `__Type` to query schemas.
```sh
Query err=Unknown type "__Type".
Query err=Unknown type "__InputValue".
Query err=Unknown type "__Type".
```
Apparently the this lib does not implement these by default. I'm thinking writing a `__Type` field for the query and then use reflect to return types.

**Comment by javiertury** [comment]
I've found a way to print the schema

```
import { buildASTSchema, printSchema } from 'graphql';
import gql from 'gql-tag';

const typeDefs = gql`
type Query {
  hello: String
}
`;
const schema = buildASTSchema(typeDefs);
const printedSchema = printSchema(buildASTSchema);
console.log(printedSchema);
```

This process can combine the utilities `makeExecutableSchema`, `mergeTypeDefs`, `stitchSchemas`, etc. from 'graphql-tools' to print your stiched schemas.

**Comment by ThomasP1988** [comment]
https://stackoverflow.com/questions/60451843/how-can-one-get-a-complete-graphql-schema-from-an-apollo-server-endpoint
